anymail/webhooks/unisender_go.py

```diff
@@ -0,0 +1,172 @@
+        "err_lost": RejectReason.OTHER,
+    }
+
+    http_method_names = ["post", "head", "options", "get"]
+
+    def get(
+        self, request: HttpRequest, *args: typing.Any, **kwargs: typing.Any
+    ) -> HttpResponse:
+        # Some ESPs verify the webhook with a GET request at configuration time
+        return HttpResponse()
+
+    def validate_request(self, request: HttpRequest) -> None:
+        """
+        How Unisender GO authenticate:
+        Hash the whole request body text and replace api key in "auth" field by this hash.
+
+        So it is both auth and encryption. Also, they hash JSON without spaces.
+        """
+        request_json = json.loads(request.body.decode("utf-8"))
+        request_auth = request_json.get("auth", "")
+        request_json["auth"] = settings.ANYMAIL_UNISENDERGO_API_KEY
+        json_with_key = json.dumps(request_json, separators=(",", ":"))
+
+        expected_auth = md5(json_with_key.encode("utf-8")).hexdigest()
+
+        if not constant_time_compare(request_auth, expected_auth):
+            raise AnymailWebhookValidationFailure(
+                f"Missing or invalid basic auth in Anymail {self.esp_name} webhook"
+            )
+
+    def parse_events(self, request: HttpRequest) -> list[AnymailTrackingEvent]:
+        request_json = json.loads(request.body.decode("utf-8"))
+        esp_events = request_json["events_by_user"][0]["events"]
+        parsed_events = [
+            self.esp_to_anymail_event(esp_event) for esp_event in esp_events
+        ]
+        return [event for event in parsed_events if event]
+
+    def esp_to_anymail_event(self, esp_event: dict) -> AnymailTrackingEvent | None:
+        if esp_event["event_name"] == "transactional_spam_block":
+            return None
+        event_data = esp_event["event_data"]
+        event_type = self.event_types.get(event_data["status"], EventType.UNKNOWN)
+        timestamp = datetime.fromisoformat(event_data["event_time"])
+        timestamp_utc = timestamp.replace(tzinfo=timezone.utc)
+        metadata = event_data["metadata"]
+        event_data["esp_name"] = self.esp_name
+        delivery_info = event_data.get("delivery_info", {})
+        unisender_delivery_status = delivery_info.get("delivery_status", "")
+        if unisender_delivery_status.startswith("err"):
+            anymail_reject_reason = self.reject_reasons.get(
+                unisender_delivery_status, RejectReason.OTHER
+            )
+        else:
+            anymail_reject_reason = ""
+
+        return AnymailTrackingEvent(
+            event_type=event_type,
+            timestamp=timestamp_utc,
+            message_id=metadata.get("message_id", None),
+            event_id=None,
+            recipient=event_data["email"],
+            reject_reason=anymail_reject_reason,
+            mta_response=delivery_info.get("destination_response", ""),
+            tags=None,
+            metadata=metadata,
+            click_url=None,
```

Comment:
Looks like `event_data.get("url")` would provide `click_url` for the "clicked" event.